File: L9.cpp
#include <iostream>
using namespace std; 

class Solution{
    public:
        bool isPalindrome(int x){
            double store =0; 
            int s = x; 
            while(x>0){
                int remainder = x%10;
                store = remainder + store*10;
                x/=10; 
            }
            if(store != s){
                return false;
            }
            return true; 
        } 
};
/*
class Solution{
    public:
        bool isPalidrome(int x){
            if(x<0){
                return false;
            }
            int s =x; 
            long store;
            while(x!=0){
                int remainder = x%10;
                store = store*10+remainder; 
                x/=10;
            }
            return store == s;
        }
};
*/

File: L9.java
class Solution{
    public  boolean isPalindrome(int x){
       if(x<0){
        return false; 
       }
       int a  = x;  //stores x
       long rev = 0; 
       while(x>0){
            int rem = x%10;
            rev = rev * 10 + rem; 
            x/=10;
       }
       return rev == a; 
    }
    
}
